arkane/ess/factory.py

```diff
@@ -0,0 +1,229 @@
+        return self.ess.load_scan_energies()
+
+    def load_scan_pivot_atoms(self):
+        """
+        Extract the atom numbers which the rotor scan pivots around.
+        """
+        return self.ess.load_scan_pivot_atoms()
+
+    def load_scan_frozen_atoms(self):
+        """
+        Extract the atom numbers where were frozen during the scan.
+        """
+        return self.ess.load_scan_frozen_atoms()
+
+    def load_negative_frequency(self):
+        """
+        Return the negative frequency from a transition state frequency calculation.
+        """
+        return self.ess.load_negative_frequency()
+
+    def get_symmetry_properties(self):
+        """
+        This method uses the symmetry package from RMG's QM module
+        and returns a tuple where the first element is the number
+        of optical isomers, the second element is the symmetry number,
+        and the third element is the point group identified.
+        """
+        coordinates, atom_numbers, _ = self.load_geometry()
+        unique_id = '0'  # Just some name that the SYMMETRY code gives to one of its jobs
+        # Scratch directory that the SYMMETRY code writes its files in:
+        scr_dir = os.path.join(os.path.abspath('.'), str('scratch'))
+        if not os.path.exists(scr_dir):
+            os.makedirs(scr_dir)
+        try:
+            qmdata = QMData(
+                groundStateDegeneracy=1,  # Only needed to check if valid QMData
+                numberOfAtoms=len(atom_numbers),
+                atomicNumbers=atom_numbers,
+                atomCoords=(coordinates, str('angstrom')),
+                energy=(0.0, str('kcal/mol'))  # Only needed to avoid error
+            )
+            # Dynamically create custom class to store the settings needed for the point group calculation
+            # Normally, it expects an rmgpy.qm.main.QMSettings object, but we don't need all of those settings
+            settings = type(str(''), (),
+                            dict(symmetryPath=str('symmetry'), scratchDirectory=scr_dir))()
+            pgc = PointGroupCalculator(settings, unique_id, qmdata)
+            pg = pgc.calculate()
+            if pg is not None:
+                optical_isomers = 2 if pg.chiral else 1
+                symmetry = pg.symmetry_number
+                logging.debug("Symmetry algorithm found {0} optical isomers and a symmetry number of {1}".format(
+                    optical_isomers, symmetry))
+            else:
+                logging.error('Symmetry algorithm errored when computing point group\nfor log file located at{0}.\n'
+                              'Manually provide values in Arkane input.'.format(self.ess.path))
+            return optical_isomers, symmetry, pg.point_group
+        finally:
+            shutil.rmtree(scr_dir)
+
+    def get_D1_diagnostic(self):
+        """
+        Returns the D1 diagnostic from the output log for certain quantum jobs.
+        """
+        return self.ess.get_D1_diagnostic()
```

Comment:
From above: If in `ESSAdapter` in `adapter.py`, I make get_D1_diagnostic and get_T1_diagnostic regular methods, rather than abstract methods, would that solve the problem or is there something else I'm not thinking of?